_posts/2023-08-25-using-kani-to-validate-security-boundaries-in-aws-firecracker.md

```diff
@@ -0,0 +1,292 @@
+---
+layout: post
+title: Using Kani to Validate Security Boundaries in AWS Firecracker
+---
+
+# Using Kani to Validate Security Boundaries in AWS Firecracker
+
+Kani is a bit-precise model checker that can verify properties about your Rust code. To learn more about Kani, check out the [Kani tutorial](https://model-checking.github.io/kani/kani-tutorial.html) and our [previous blog posts](https://model-checking.github.io/kani-verifier-blog/).
+
+For the past 10 months, Felipe Monteiro, an Applied Scientist on the [Kani](https://github.com/model-checking/kani/) team and Patrick Roy, a Software Development Engineer from the [AWS Firecracker](https://github.com/firecracker-microvm/firecracker) team, collaborated to develop Kani harnesses that verify safety-critical properties in the Firecracker Virtual Machine Monitor. Firecracker is an open source project written in Rust which uses the Linux Kernel-based Virtual Machine (KVM) to create and manage microVMs. Firecracker has a minimalist design which allows fast (~150ms) microVM start-up time, secure multi-tenancy of microVMs on the same host and memory/CPU over-subscription. Firecracker is currently used in production by AWS Lambda, AWS Fargate and parts of AWS Analytics to build their service platforms. 
+
+As a result of this collaboration, the Firecracker team is now running multiple verification suites in their continuous integration pipelines, ensuring that all checked properties of critical systems are upheld on every code change.
```

Comment:
```suggestion
As a result of this collaboration, the Firecracker team is now running multiple verification harnesses in their continuous integration pipelines, ensuring that all checked properties of critical components are upheld on every code change.
```

Reply:
Suggestion: Maybe we could include a bit more data here? For instance, "Firecracker runs 26 Kani harnesses that covers 3 critical modules, which takes approximately X seconds to complete".